prometheus/histogram.go

```diff
@@ -742,6 +761,8 @@ func (h *histogram) Observe(v float64) {
 	h.observe(v, h.findBucket(v))
 }
 
+// ObserveWithExemplar should not be called in high-frequency settings,
+// since it isn't lock-free for native histograms with configured exemplars.
```

Comment:
Slight clarification:

ObserveWithExemplar should not be called in a high-frequency setting for a native histogram with configured exemplars. For this case, the implementation isn't lock-free and might suffer from lock contention.

Reply:
Done